ui/test/nuxeo-document-comment-thread.test.js

```diff
@@ -0,0 +1,392 @@
+        await flush();
+
+        expect(isElementVisible(element.shadowRoot.querySelector('span.more-content'))).to.be.false;
+      });
+
+      test('Should display only 10 comments when more than 10 comments are returned from server', async () => {
+        const comment = {
+          'entity-type': 'comment',
+          parentId: 'doc-id',
+          id: 'comment-id',
+          numberOfReplies: 0,
+          author: 'John Doe',
+          creationDate: '2019-12-09',
+          text: 'This is my testing comment',
+        };
+
+        server.respondWith('get', '/api/v1/id/doc-id/@comment/', {
+          'entity-type': 'comments',
+          entries: Array(10).fill(comment),
+          totalSize: 12,
+        });
+
+        element._refresh();
+        await flush();
+
+        expect(element.shadowRoot.querySelectorAll('nuxeo-document-comment').length).to.equal(10);
+      });
+
+      test('Should display "load more" link when more than 10 comments are returned from server', async () => {
+        const comment = {
+          'entity-type': 'comment',
+          parentId: 'doc-id',
+          id: 'comment-id',
+          numberOfReplies: 0,
+          author: 'John Doe',
+          creationDate: '2019-12-09',
+          text: 'This is my testing comment',
+        };
+
+        server.respondWith('get', '/api/v1/id/doc-id/@comment/', {
+          'entity-type': 'comments',
+          entries: Array(10).fill(comment),
+          totalSize: 12,
+        });
+
+        element._refresh();
+        await flush();
+
+        expect(isElementVisible(element.shadowRoot.querySelector('span.more-content'))).to.be.true;
+      });
+
+      suite('Reconciliation', () => {});
+    });
+  });
+
+  suite('Events', () => {
+    test('Should remove comment from list when deletion event is received', async () => {
+      const commentOne = {
+        'entity-type': 'comment',
+        parentId: 'doc-id',
+        id: 'comment-id-one',
+        numberOfReplies: 0,
+        author: 'John Doe',
+        creationDate: '2019-12-09',
+        text: 'This is my testing comment',
+      };
+
+      const commentTwo = {
+        'entity-type': 'comment',
+        parentId: 'doc-id',
+        id: 'comment-id-two',
+        numberOfReplies: 0,
+        author: 'Mary Poppins',
+        creationDate: '2019-12-19',
+        text: 'Cool!',
+      };
+
+      server.respondWith('get', '/api/v1/id/doc-id/@comment/', {
+        'entity-type': 'comments',
+        entries: [commentOne, commentTwo],
+        totalSize: 2,
+      });
+
+      element._refresh();
+      await flush();
+
+      const comment = element.shadowRoot.querySelectorAll('nuxeo-document-comment')[0];
+      comment.dispatchEvent(
+        new CustomEvent('delete-comment', { bubbles: true, composed: true, detail: { commentId: 'comment-id-two' } }),
+      );
+      await flush();
+
+      expect(element.shadowRoot.querySelectorAll('nuxeo-document-comment').length).to.equal(1);
```

Comment:
nit: shouldn't we assert the ID instead of the length?

Reply:
The ID is gone since the comment was deleted.
We can, maybe, assert that the comment left is not the one we removed 🤷‍♂